tests/test_frame.py

```diff
@@ -985,6 +985,20 @@ def test_types_groupby_any() -> None:
     )
 
 
+def test_types_groupby_iter() -> None:
+    df = pd.DataFrame(data={"col1": [1, 1, 2], "col2": [3, 4, 5]})
+    series_groupby = pd.Series([True, True, False], dtype=bool)
+    first_group = next(iter(df.groupby(series_groupby)))
+    check(
+        assert_type(first_group[0], "bool"),
+        bool,
+    )
+    check(
+        assert_type(first_group[1], "pd.DataFrame"),
```

Comment:
You don't need quotes around the types in the two `assert_type()` statements, because those are valid types at runtime.